### ISSUE TITLE
iOS 13 - Cannot install app on real device

### ISSUE THREAD

**Issue by hardik-plume** [appium/appium#13246]
## The problem

ios-deploy works manually but not through appium

## Environment

Appium version (or git revision) that exhibits the issue: 1.15.0-beta.1
Last Appium version that did not exhibit the issue (if applicable): 1.13
Desktop OS/version used to run Appium: mac OS 10.14.6
Node.js version (unless using Appium.app|exe): v12.12.0
Npm or Yarn package manager: 6.11.3
Mobile platform/version under test: iOS 13 beta
Real device or emulator/simulator: iPhone Xs Max
Appium CLI or Appium.app|exe: cli

## Link to Appium logs

```[BaseDriver] Number capability passed in as string. Functionality may be compromised.
[BaseDriver] Capability 'wdaLocalPort' changed from string ('') to integer (NaN). This may cause unexpected behavior
[BaseDriver] The following capabilities were provided, but are not recognized by Appium:
[BaseDriver]   unicodeKeyboard
[BaseDriver]   resetKeyboard
[XCUITest] 'platformVersion' capability ('') is not a valid version number. Consider fixing it or be ready to experience an inconsistent driver behavior.
[BaseDriver] Session created with session id: 3a6af3db-85e2-4442-8fc1-5ad417ddb88d
[debug] [XCUITest] Current user: 'hardik'
[debug] [XCUITest] Available devices: 00008020-001551003CC2002E
[debug] [XCUITest] Creating iDevice object with udid '00008020-001551003CC2002E'
[XCUITest] Getting the platformVersion from the phone since it was not specified in the capabilities
[XCUITest] Determining device to run tests on: udid: '00008020-001551003CC2002E', real device: true
[XCUITest] Normalized platformVersion capability value '13.0.0' to '13.0'
[debug] [BaseDriver] Event 'xcodeDetailsRetrieved' logged at 1568664592524 (13:09:52 GMT-0700 (PDT))
[BaseDriver] Using local app '/Users/hardik/build/generic/plume/Plume – DevIntegration.app'
[debug] [BaseDriver] Event 'appConfigured' logged at 1568664592528 (13:09:52 GMT-0700 (PDT))
[debug] [XCUITest] Checking whether app '/Users/hardik/build/generic/plume/Plume – DevIntegration.app' is actually present on file system
[debug] [XCUITest] App is present
[debug] [iOS] Getting bundle ID from app '/Users/hardik/build/generic/plume/Plume – DevIntegration.app': 'com.plumewifi.plume.devintegration'
[debug] [BaseDriver] Event 'resetStarted' logged at 1568664592531 (13:09:52 GMT-0700 (PDT))
[debug] [XCUITest] Reset: running ios real device reset flow
[debug] [BaseDriver] Event 'resetComplete' logged at 1568664592531 (13:09:52 GMT-0700 (PDT))
[debug] [XCUITest] Crash reports root '/Users/hardik/Library/Logs/CrashReporter/MobileDevice/Plume’s iPhone XS Max' does not exist. Got nothing to gather.
[debug] [BaseDriver] Event 'logCaptureStarted' logged at 1568664592786 (13:09:52 GMT-0700 (PDT))
[XCUITest] Setting up real device
[debug] [XCUITest] Verifying application platform
[debug] [XCUITest] CFBundleSupportedPlatforms: ["iPhoneOS"]
[debug] [XCUITest] Installing '/Users/hardik/build/generic/plume/Plume – DevIntegration.app' on device with UUID '00008020-001551003CC2002E'...
[XCUITest] Error was thrown during the installation process
[XCUITest] TypeError: Path must be a string. Received undefined
[XCUITest]     at assertPath (path.js:28:11)
[XCUITest]     at Object.join (path.js:1233:7)
[XCUITest]     at Object.join [as walkDir] (/usr/local/lib/node_modules/appium/node_modules/appium-support/lib/fs.js:76:29)
[XCUITest]     at <anonymous>
[XCUITest] Error: Could not install app: 'Path must be a string. Received undefined'
[XCUITest]     at IOSDeploy.install (/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/ios-deploy.js:41:13)
[XCUITest]     at <anonymous>
[debug] [BaseDriver] Event 'newSessionStarted' logged at 1568664593213 (13:09:53 GMT-0700 (PDT))
[debug] [W3C] Encountered internal error running command: Error: Could not install app: 'Path must be a string. Received undefined'
[debug] [W3C]     at IOSDeploy.install (/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/ios-deploy.js:41:13)
[debug] [W3C]     at <anonymous>
[HTTP] <-- POST /wd/hub/session 500 1181 ms - 649
```

**Comment by KazuCocoa** [comment]
Could you try it out with Node 10+ instead of 8.15.0?

**Comment by hardik-plume** [comment]
@KazuCocoa i read it on some other issue and then updated to node 12 and reinstalled appium beta. No luck

**Comment by hardik-plume** [comment]
here are the logs with node 12.12:

```[BaseDriver] Number capability passed in as string. Functionality may be compromised.
[BaseDriver] Capability 'wdaLocalPort' changed from string ('') to integer (NaN). This may cause unexpected behavior
[BaseDriver] The following capabilities were provided, but are not recognized by Appium:
[BaseDriver]   unicodeKeyboard
[BaseDriver]   resetKeyboard
[BaseDriver] Session created with session id: 02709f11-4178-4e4b-8c0d-3bbb71e5747c
[debug] [XCUITest] Current user: 'hardik'
[debug] [XCUITest] Available devices: 00008020-001551003CC2002E
[debug] [XCUITest] Creating iDevice object with udid '00008020-001551003CC2002E'
[XCUITest] Determining device to run tests on: udid: '00008020-001551003CC2002E', real device: true
[debug] [BaseDriver] Event 'xcodeDetailsRetrieved' logged at 1568681420310 (17:50:20 GMT-0700 (Pacific Daylight Time))
[BaseDriver] Using local app '/Users/hardik/build/generic/plume/Plume–DevIntegration.app'
[debug] [BaseDriver] Event 'appConfigured' logged at 1568681420315 (17:50:20 GMT-0700 (Pacific Daylight Time))
[debug] [XCUITest] Checking whether app '/Users/hardik/build/generic/plume/Plume–DevIntegration.app' is actually present on file system
[debug] [XCUITest] App is present
[debug] [iOS] Getting bundle ID from app '/Users/hardik/build/generic/plume/Plume–DevIntegration.app': 'com.plumewifi.plume.devintegration'
[debug] [BaseDriver] Event 'resetStarted' logged at 1568681420318 (17:50:20 GMT-0700 (Pacific Daylight Time))
[debug] [XCUITest] Reset: running ios real device reset flow
[debug] [BaseDriver] Event 'resetComplete' logged at 1568681420319 (17:50:20 GMT-0700 (Pacific Daylight Time))
[debug] [XCUITest] Crash reports root '/Users/hardik/Library/Logs/CrashReporter/MobileDevice/Plume’s iPhone XS Max' does not exist. Got nothing to gather.
[debug] [BaseDriver] Event 'logCaptureStarted' logged at 1568681420537 (17:50:20 GMT-0700 (Pacific Daylight Time))
[XCUITest] Setting up real device
[debug] [XCUITest] Verifying application platform
[debug] [XCUITest] CFBundleSupportedPlatforms: ["iPhoneOS"]
[debug] [XCUITest] Reset requested. Removing app with id 'com.plumewifi.plume.devintegration' from the device
[debug] [XCUITest] Installing '/Users/hardik/build/generic/plume/Plume–DevIntegration.app' on device with UUID '00008020-001551003CC2002E'...
[XCUITest] Error was thrown during the installation process
[XCUITest] TimeoutError: operation timed out
[XCUITest]     at afterTimeout (/usr/local/lib/node_modules/appium/node_modules/bluebird/js/release/timers.js:46:19)
[XCUITest]     at Timeout.timeoutTimeout [as _onTimeout] (/usr/local/lib/node_modules/appium/node_modules/bluebird/js/release/timers.js:76:13)
[XCUITest]     at listOnTimeout (internal/timers.js:531:17)
[XCUITest]     at processTimers (internal/timers.js:475:7)
[XCUITest] Error: Could not install app: 'operation timed out'
[XCUITest]     at IOSDeploy.install (/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/ios-deploy.js:41:13)
[debug] [BaseDriver] Event 'newSessionStarted' logged at 1568681421167 (17:50:21 GMT-0700 (Pacific Daylight Time))
[debug] [W3C] Encountered internal error running command: Error: Could not install app: 'operation timed out'
[debug] [W3C]     at IOSDeploy.install (/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/ios-deploy.js:41:13)
[HTTP] <-- POST /wd/hub/session 500 1431 ms - 585
```

**Comment by mykola-mokhnach** [comment]
We track this problem in https://github.com/appium/appium/issues/13180
